### ISSUE TITLE
Data Plugin and large files

### ISSUE THREAD

**Issue by sonOfRa** [Novik/ruTorrent#1327]
When downloading large files (larger than 2GiB) with the Data plugin, the downloaded data is corrupted. The corruption starts at exactly 2147483647 bytes, which is the overflow boundary for a signed 32bit integer.

There is an attempt to handle this overflow in the current sendFile utility, however it is ineffective. 

https://github.com/Novik/ruTorrent/blob/4d7d5f5ff42440e34c61977fac2a0fd5a75a521c/php/util.php#L610

First, a check is made whether the ints are 32/64 bits, and then another check whether the size is too large. However, if the int is only 32 bits, the value that's being checked is already causing an overflow, thus causing the check to never return true, thus never executing the cat-workaround.

I worked around this by installing mod_xsendfile, so it works for now, but maybe this overflow error is worth looking into.

**Comment by sonOfRa** [comment]
So, this didn't let me sleep, and I did some experimentation.

While the handling for systems with small ints is also problematic that wasn't the problem at hand (and it couldn't have been, I'm on a 64bit server, PHP_INT_SIZE is 8, and stat was reporting the correct file size).

I followed the control flow path for my case, and the code ended up at the readfile() bit here: https://github.com/Novik/ruTorrent/blob/4d7d5f5ff42440e34c61977fac2a0fd5a75a521c/php/util.php#L638

So it had to be something to do with readfile. I quickly wrote a simple poc for the command line, and came to the conclusion that readfile() simply isn't suitable for big files. Here's the result:

https://gist.github.com/sonOfRa/37bd8e4671040b76db02d1fed19c6760

The garbling of the output file seems to be deterministic, I got the exact same md5sum for the output file after dropping pagecache, dentries and inode cache
